control/strategy/pool.go

```diff
@@ -347,21 +359,45 @@ func (p *pool) SubscriptionCount() int {
 }
 
 // SelectAP selects an available plugin from the pool
-func (p *pool) SelectAP(taskID string) (SelectablePlugin, serror.SnapError) {
+func (p *pool) SelectAP(opts SelectorValues) (AvailablePlugin, serror.SnapError) {
```

Comment:
> added SelectorValues struct for keeping "selectors" for available strategies in one place. Easy to extend, used to pass taskID and config to pool.SelectAP() method

Multiple arguments vs options struct - where do we draw the line? 
For me it generally comes down to code readability.  In this case I feel that `SelectorValues` decreases code readability.  `pool.Select` accepting the task id and configs is more readable and wouldn’t require the anonymous function in SelectAP.